docs/source/setup_virtualenv.rst

```diff
@@ -0,0 +1,181 @@
+Set up a virtual environment
+============================
+
+First, make sure you have Python installed:
+
+-  :doc:`installing_python`
+
+
+Set up a virtualenv
+-------------------
+
+Set up a virtual environment by running the following commands:
+
+OSX
+~~~
+
+If using Python 2.7 and pip:
+::
+
+
+    pip install virtualenv
+    pip install virtualenvwrapper
+    export WORKON_HOME=$HOME/.virtualenvs
+    mkdir -p $WORKON_HOME
+    export VIRTUALENVWRAPPER_PYTHON=$(which python)
+    source $(which virtualenvwrapper.sh)
+
+If using Python 3.x and pip3 (Python 3.7 in this example):
+::
+
+
+    pip3 install virtualenv
+    pip3 install virtualenvwrapper
+    export WORKON_HOME=$HOME/.virtualenvs
+    mkdir -p $WORKON_HOME
+    export VIRTUALENVWRAPPER_PYTHON=$(which python3.7)
+    source $(which virtualenvwrapper.sh)
+
+
+Now create the virtual environment using:
+::
+
+
+    mkvirtualenv dlgr_env --python <specify_your_python_path_here>
+
+
+Examples:
+
+Using homebrew installed Python 3.7:
+::
+
+
+    mkvirtualenv dlgr_env --python /usr/local/bin/python3.7
+
+Using Python 2.7:
+::
+
+
+    mkvirtualenv dlgr_env --python /usr/bin/python
+
+
+These commands use ``pip/pip3``, the Python package manager, to install two
```

Comment:
Hmm... I think I might explain the "why" rather than the "how" here: "Virtualenvwrapper provides an easy way to switch between virtual environments by simply typing `workon [virtual environment name]`." Etc.

The "how" gets pretty dense pretty fast.

Reply:
True

Reply:
Ah the refactored docs make changes so much quicker :) :1st_place_medal: :horns